src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -119,6 +126,10 @@ type PromWriteHandler struct {
 	nowFn                  clock.NowFn
 	instrumentOpts         instrument.Options
 	metrics                promWriteMetrics
+
+	// Counting the number of times of "literal is too long" error for log sampling purposes.
+	numLiteralIsTooLong     int
+	numLiteralIsTooLongLock sync.RWMutex
```

Comment:
Why not use atomic int?

Reply:
https://github.com/m3db/m3/pull/3801/commits/562dcf01ed4e27734749840e3218f1fb87d748c2